### ISSUE TITLE
[BUG] OpenSearch gets stuck opening indices since homebrew OpenJDK update

### ISSUE THREAD

**Issue by unilynx** [opensearch-project/OpenSearch#12346]
### Describe the bug

First - I know opensearch currently isn't supported on M3s and homebrew is not an official release, but it's still really convenient to have and I hope I can get some pointers here.

since homebrew updated its openjdk version (to 21.0.1) - https://github.com/Homebrew/homebrew-core/commit/1c027971bffc50813dec26940d6633b9762f785e - opensearch has been broken for me and a few colleagues when you're in the 100+ index range. 

when starting opensearch it will only open part of the existing indices (the rest stays red). how many indices go green differs per startup. often just 24 out of 229, but also eg. 43 or 142. setting `-Elogger.level=DEBUG` seems to open more (but still not all) indices after restart, so I suspect a locking/timing issue.

Indices that stay in red all debug-log messages like this at startup

- throttling allocation
- found 1 allocation candidates of
  - (the above two messages repeat a few times)
- allocating
- found 1 allocation candidates
- allocating

and then the messages stop for that index. I notice that a log line with allocating appears twice which is suspect because indices that *do* make it to green log this:

- throttling allocation
- found 1 allocation candidates of
  - (the above two messages repeat a few times)
- allocating
- creating index
- creating Index ..shards [1]/[0] - reason [CREATE_INDEX]

just one 'allocating' and after that the index opens and makes it to green.

downgrading openjdk fixes it and allows me to open all indices, so the data itself doesn't appear to have been corrupted. I can reproduce this on muiltiple machines and wiping the data dirs completely doesn't make a difference.

### Related component

Search

### To Reproduce

Please see above

### Expected behavior

All indices should go green like they do when I downgrade openjdk

### Additional Details

**Plugins**
analysis-icu

**Host/Environment (please complete the following information):**
 - OS: Sonoma 14.2.1 (23C71)
- Hardware: Macbook M3 Pro

**Comment by andrross** [comment]
Just one quick sanity check...is the JVM arg `-Djava.security.manager=allow` being passed at startup? It should be, and if it wasn't I think the JVM would exit, but that's one obvious difference that JDK21 requires.

**Comment by andrross** [comment]
Also which version of OpenSearch are you using and how are you starting it (i.e. inside a development workspace via `.gradlew run`, from a tarball, from docker, etc)?

**Comment by reta** [comment]
@unilynx just to clarify, 2.11.1 does not work with `JDK 21.0.2`, could you please confirm the JVM version that is being used on startup? (PS: 2.12.0 fixes the  `JDK 21.0.2` issue)

**Comment by unilynx** [comment]
I currently pinned my JDK to 21.0.1, and then opensearch works

```
moe:~ arnold$ /opt/homebrew/bin/opensearch --version
warning: no-jdk distributions that do not bundle a JDK are deprecated and will be removed in a future release
WARNING: A terminally deprecated method in java.lang.System has been called
WARNING: System::setSecurityManager has been called by org.opensearch.bootstrap.OpenSearch (file:/opt/homebrew/Cellar/opensearch/2.11.1/libexec/lib/opensearch-2.11.1.jar)
WARNING: Please consider reporting this to the maintainers of org.opensearch.bootstrap.OpenSearch
WARNING: System::setSecurityManager will be removed in a future release
Version: 2.11.1, Build: tar/unknown/2023-11-29T21:16:04Z, JVM: 21.0.1
```

> just to clarify, 2.11.1 does not work with `JDK 21.0.2` 

Sounds like homebrew shouldn't have upgraded the JDK then (but they probably have more dependents than just opensearch on the JDK)

If I unpin and upgrade the openjdk:

```
moe:~ arnold$ /opt/homebrew/bin/opensearch --version
warning: no-jdk distributions that do not bundle a JDK are deprecated and will be removed in a future release
WARNING: A terminally deprecated method in java.lang.System has been called
WARNING: System::setSecurityManager has been called by org.opensearch.bootstrap.OpenSearch (file:/opt/homebrew/Cellar/opensearch/2.11.1/libexec/lib/opensearch-2.11.1.jar)
WARNING: Please consider reporting this to the maintainers of org.opensearch.bootstrap.OpenSearch
WARNING: System::setSecurityManager will be removed in a future release
Version: 2.11.1, Build: tar/unknown/2023-11-29T21:16:04Z, JVM: 21.0.2
```

So the above would describe the 'broken' situation when I encountered this issue. 

Anyway - the good news is: Homebrew updated opensearch to 2.12.0

```
Version: 2.12.0, Build: tar/unknown/2024-02-20T00:41:43Z, JVM: 21.0.2
```

and the issue now seems resolved for me. tried a couple of restarts, all indices now properly open. 

The issue is now resolved for me. I'll see what happens in practice when this:

```
warning: no-jdk distributions that do not bundle a JDK are deprecated and will be removed in a future release
```

goes into effect and whether Homebrew will then ship with a bundled JDK which should fix similar issues in the future. Or if not we'll have to look into just using the targz file directly. Think I'm going to have to figure out who in Homebrew feels responsible for the opensearch formula and whether it's possibly to bundle Opensearch's preferred JDK instead of unbundling it, but it might take Opensearch actually dropping support for unbundling.

TL;DR: Resolved
